### PR TITLE
[2.6] backport of ASM 6.2.0 to 2.6 branch

### LINKED ISSUE #164: 2.6.5 release breaks JDK7 compatibility
commit a49fe7bb34b881fdb86dbd10cd7d9431844c2117 upgrades asm_5.0.1.v201405080102 to asm_6.0.0.v201702131300. This commit also resides within 2.6.5 release which breaks JDK7 compatibility since asm_6.0.0.v201702131300 requires JDK8 at least

### PR DESCRIPTION
**Pull request by lukasj**
this also fixes #164

### CODE REVIEW

**Comment by lukasj** [comment]
test-lrg passed after the change on my end

File: plugins/org.eclipse.persistence.asm/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
+ l;version="6.2.0";uses:="org.xml.sax.helpers,org.eclipse.persistence.
+ internal.libraries.asm,org.xml.sax,javax.xml.transform"
 Bundle-Name: EclipseLink ASM
 Created-By: 1.8.0_121 (Sun Microsystems Inc.)
 HK2-Bundle-Name: org.eclipse.persistence:org.eclipse.persistence.asm
 Bundle-RequiredExecutionEnvironment: JavaSE-1.7
 Implementation-Vendor: Eclipse.org - EclipseLink Project
 Bundle-Vendor: Eclipse.org - EclipseLink Project
-Bundle-Version: 6.0.0.v201702131300
+Bundle-Version: 6.2.0.qualifier
```
[the remark this answers is not in the document]
Comment (lukasj) [reply plugins/org.eclipse.persistence.asm/META-INF/MANIFEST.MF]: datestamp gets there during the build

File: plugins/org.eclipse.persistence.asm/src/org/eclipse/persistence/internal/libraries/asm/AnnotationVisitor.java [hunk 2]
```diff
@@ -1,170 +1,150 @@
...
-    /**
-     * The ASM API version implemented by this visitor. The value of this field
-     * must be one of {@link Opcodes#ASM4}, {@link Opcodes#ASM5} or {@link Opcodes#ASM6}.
-     */
-    protected final int api;
+  /**
+   * The ASM API version implemented by this visitor. The value of this field must be one of {@link
+   * Opcodes#ASM4}, {@link Opcodes#ASM5}, {@link Opcodes#ASM6} or {@link Opcodes#ASM7_EXPERIMENTAL}.
+   */
+  protected final int api;
```
[the remark this answers is not in the document]
Comment (lukasj) [reply plugins/org.eclipse.persistence.asm/src/org/eclipse/persistence/internal/libraries/asm/AnnotationVisitor.java]: that is how it is in asm 6.2.0. I did no reformat nor any other changes in asm sources; they should 1:1 match to original

File: sdo/eclipselink.sdo.test/antbuild.properties [hunk 3]
```diff
@@ -37,7 +37,7 @@ classes.dir=classes
 resource.dir=resource
 report.dir=reports
 
-asm=org.eclipse.persistence.asm_6.0.0.v201702131300.jar
+asm.jar=org.eclipse.persistence.asm_6.2.0.v201808041654.jar
```
[the remark this answers is not in the document]
Comment (lukasj) [reply sdo/eclipselink.sdo.test/antbuild.properties]: honestly - I found it's actually not used, so did not pay extra attention to this but you're right, the change is not correct. It's error in merge...

[the remark this answers is not in the document]
Comment (lukasj) [reply sdo/eclipselink.sdo.test/antbuild.properties]: should be fixed

File: plugins/org.eclipse.persistence.asm/about.html [hunk 4]
```diff
@@ -32,41 +32,61 @@ <h2>Third Party Content</h2>
 <h3><a name="ASM" id="ASM"></a>ASM v6.0.0.ALPHA</h3>
```
[the remark this answers is not in the document]
Comment (lukasj) [reply plugins/org.eclipse.persistence.asm/about.html]: good catch, should be fixed

**Comment by lukasj** [comment]
try it now, see https://ci.eclipse.org/eclipselink/job/eclipselink-update-compdeps/43/console

**Comment by dazey3** [comment]
Yep, it works now +  https://github.com/eclipse-ee4j/eclipselink/pull/210 ! Thanks @lukasj

**Comment by dazey3** [comment]
I am attempting to port this change to 2.6_WAS branch, but I am getting the same exception I mentioned above. I was getting that exception on BOTH 2.6 and 2.6_WAS, but you must have done something to fix that issue. I'd like to get that fix for 2.6_WAS too...

**Comment by lukasj** [comment]
see my comment in #209 - rebuild compdeps

**Pull request merged**